### PR TITLE
Benchmark scripts for MemoryFootprint and TPC-H with/without mmap performance

### PR DESCRIPTION
**Pull request by phoeinx**
This PR adds benchmarking scripts used for our end presentation and the report.

They i.a. support measuring:
- measuring memory peaks of benchmarks in hyrise
- measuring performance of mmap-based vs. master-hyrise with varying clients and core configurations
- measuring mmap-based hyrise with limited memory using cgrups
- scripts to run multiple benchmarks sequentially (using python subprocess module)
- visualising memory peaks of benchmarks in hyrise in various formats
- visualising all other measurements

### CODE REVIEW

File: scripts/cgroups/run_cgroup_limited_benchmark.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import os
+import subprocess
+import time
+import signal
+
+GB = 1024 * 1024 * 1024
```
Comment (Bouncner) [inline scripts/cgroups/run_cgroup_limited_benchmark.py:6]: 1024 is GiB. 1000 is GB.

Comment (phoeinx) [reply scripts/cgroups/run_cgroup_limited_benchmark.py:6]: I know, but I've seen it used somewhat interchangeably. But you would argue to be precise here?

Comment (Bouncner) [reply scripts/cgroups/run_cgroup_limited_benchmark.py]: Naming the variable `GiB` would satisfy me. :)

**Comment by phoeinx** [comment]
@Bouncner @mweisgut @klauck 

What type of clean up would you prefer for those scripts for the submission?
I would propose moving them to a `mp2023` folder to make it easier to see which scripts were added by us - this is also necessary in our master. 
Anything else? Those scripts are as of now mostly pretty specific for our measurements (except the memory peak script for hyrise) and perhaps the script to run multiple scripts sequentially, therefore I don't expect much further usage.

Thank you!

**Comment by Bouncner** [comment]
> @Bouncner @mweisgut @klauck
> 
> What type of clean up would you prefer for those scripts for the submission? I would propose moving them to a `mp2023` folder to make it easier to see which scripts were added by us - this is also necessary in our master. Anything else? Those scripts are as of now mostly pretty specific for our measurements (except the memory peak script for hyrise) and perhaps the script to run multiple scripts sequentially, therefore I don't expect much further usage.
> 
> Thank you!

Sounds good to me.

File: scripts/mp-2023/benchmark_multicore_over_clients.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+num_cores = [24, 48]
+
+warmup = [True, False]
+
+num_clients = [1, 2, 4, 8, 16, 24, 32, 48]
+
+for num_core, num_client, do_warmup in itertools.product(num_cores, num_clients, warmup):
+
+    #neccessary for mmap-measurements, doesn't disturb other benchmarks
+    os.system("sudo rm -rf *.bin")
```
Comment (rrcomtech) [inline scripts/mp-2023/benchmark_multicore_over_clients.py:22]: Looks frightening, but ok.

File: src/lib/storage/storage_manager.cpp [hunk 3]
```diff
@@ -505,9 +504,9 @@ FILE_HEADER StorageManager::_read_file_header(const std::string& filename) const
   auto file_header = FILE_HEADER{};
   auto fd = int32_t{};
 
-  Assert((fd = open((_persistence_directory + filename).c_str(), O_RDONLY)) >= 0, "Open error");
-  auto* persisted_header =
-      reinterpret_cast<uint32_t*>(mmap(NULL, _file_header_bytes, PROT_READ, MAP_PRIVATE, fd, off_t{0}));
+  Assert((fd = open(filename.c_str(), O_RDONLY)) >= 0, "Open error");
+  auto* map_pointer = mmap(NULL, _file_header_bytes, PROT_READ, MAP_PRIVATE, fd, off_t{0});
+  auto* persisted_header = reinterpret_cast<uint32_t*>(map_pointer);
```
Comment (rrcomtech) [inline src/lib/storage/storage_manager.cpp:509]: Why do we have a change here? It seems to me, that this should not have been changed.

Comment (phoeinx) [reply src/lib/storage/storage_manager.cpp:509]: Thank you for the comment - yep I think this is an artefact of the rebase.

**Review by rrcomtech — APPROVED** [review]

**Pull request merged**